### PR TITLE
Update Clippy

### PR DESCRIPTION
**Pull request by Jarcho**
r? @Manishearth

### CODE REVIEW

**Comment by Jarcho** [comment]
Why does this even compile successfully in the clippy repo? The `sync` bound on lint passes isn't new.

**Comment by Manishearth** [comment]
We pin a nightly https://github.com/rust-lang/rust-clippy/blob/master/rust-toolchain

**Comment by Manishearth** [comment]
@bors delegate+ 


feel free to r=self updates

File: src/tools/clippy/clippy_utils/src/macros.rs [hunk 1]
```diff
@@ -332,121 +337,495 @@ fn is_assert_arg(cx: &LateContext<'_>, expr: &Expr<'_>, assert_expn: ExpnId) ->
...
+    ) -> Option<Self> {
+        Some(match count {
+            rpf::Count::CountIs(val) => Self::Is(val, span_from_inner(values.format_string_span, inner?)),
+            rpf::Count::CountIsName(name, span) => Self::Param(FormatParam::new(
+                FormatParamKind::Named(Symbol::intern(name)),
+                position?,
+                span,
+                values,
+            )?),
+            rpf::Count::CountIsParam(_) => {
```
Comment (Alexendoo) [inline src/tools/clippy/clippy_utils/src/macros.rs:647]: For #101000 it'll be fine to tack it on here as `| ...CountIsStar(_)`, can sort it out fully later in the clippy repo if there's anywhere that can use the info

Comment (Jarcho) [reply src/tools/clippy/clippy_utils/src/macros.rs:647]: Sounds good.

Comment (Jarcho) [reply src/tools/clippy/clippy_utils/src/macros.rs]: There's an error in the parsing somewhere. All the tests with `*` as the index aren't working.

Comment (Alexendoo) [reply src/tools/clippy/clippy_utils/src/macros.rs]: Ah that'll be https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/write.rs#L543-L564, can apply the same treatment there

**Pull request merged**